### ISSUE TITLE
Virtual Assistant - Telemetry isn't working

### ISSUE THREAD

**Issue by mekinney** [microsoft/botframework-solutions#1341]
## Project - Virtual Assistant
**Language**: C#

## Description
Telemetry Custom Events aren't working in the virtual assistant beyond the basic BotMessageSend and BotMessageReceived events.  None of the enhanced reporting events described in the Analytics section are captured.

The test case a generic VirtualAssistant with no skills.

This is likely a configuration issue, but I can't seem to find information about this.  Previously, telemetry was capture through recognizers, but that appears to have changed. 

## To Reproduce
Create a VA bot with template and configure telemetry, and run it.
The TelemetryClient is configured properly because my own custom telemetry events are logged properly.

A symptom might be that the TelemetryClient property appears to be reset to NullBotTelemetryClient for the MainDialog sometime after the constructor is called.

Snippet from MainDialog Constructor.  Saving telemetryClient in my own local plus displaying to App Insights.

```csharp
TelemetryClient = telemetryClient;
_telemetryClient = telemetryClient;
TelemetryClient.TrackEvent($"MainDialog Constructor - Telemetry Client {TelemetryClient.ToString()}");
```
Snippet from MainDialog RouteAsync.  Note from screen capture that TelemetryClient has been reset to NullTelemetry.
```
_telemetryClient.TrackEvent($"MainDialog RouteAsync _telemetryClient {_telemetryClient.ToString()}");
 _telemetryClient.TrackEvent($"MainDialog RouteAsync TelemetryClient {TelemetryClient.ToString()}");
```

Snippet from DialogBot Constructor to see if it's being extracted properly there (and it is):
```csharp
_telemetryClient = serviceProvider.GetService<IBotTelemetryClient>() ?? throw new ArgumentNullException(nameof(IBotTelemetryClient));
_telemetryClient.TrackEvent($"DialogBot Constructor - _telemetryClient: {_telemetryClient.ToString()}");
```
Here is the App Insights produced:
![Capture](https://user-images.githubusercontent.com/7552362/57542123-9f878400-7305-11e9-876e-5f124e155577.JPG)

## Expected behavior
Bot Telemetry should be produced.

**Comment by ryanisgrig** [comment]
Thanks @mekinney, the `TelemetryClient` needs to be passed in the LUIS and QnA recognizers to properly log, you can take a look at #1354 for a sample.

@enzocanoo can you verify if the Typescript version needs this update as well?

**Comment by darrenj** [comment]
@mekinney - good spot, this should have been wired up and we'll publish a new VSIX shortly. If you move the above changes across you should be good to go.

**Comment by enzocano** [comment]
Hey @ryanlengel, we added the `Telemetry` implementation for TypeScript in #1322 and fixed the `QnA` implementation with #1357.

**Comment by mekinney** [comment]
Thanks.  This fixed it.
